sequencer/src/api/fs.rs

```diff
@@ -50,15 +50,12 @@ impl SequencerDataSource for DataSource {
         // one block into memory, and rarely very many.
         let blocks: Vec<_> = self
             .get_block_range(from_block..)
-            .await?
+            .await
             .enumerate()
             .collect()
             .await;
-        for (i, block) in blocks {
-            let Ok(block) = block else {
-                tracing::warn!("missing block {}, index may be out of date", from_block + i);
-                continue;
-            };
+        for (_, block) in blocks {
+            let block = block.await;
```

Comment:
@jbearer is this change acceptable? Seems that it's no longer possible to have a missing block. Seems good but I want to be sure I've not overlooked something.

Reply:
Same as above, if the block is missing this will block until it can be retrieved. I think it would be better here to use `let Ok(block) = block.try_resolve() else {`, to mirror the previous behavior.